### ISSUE TITLE
Combobox items read back as "blank" by NVDA and JAWS

### ISSUE THREAD

**Issue by mflewellingesri** [Esri/calcite-components#4388]
### Actual Behavior

Navigating between form components, the screen reader reads out "list item" but not the label or role information. Radio group seems to be working as expected

### Expected Behavior

navigatign to each form component, the screen reader should read the label and appropriate role information

### Reproduction Sample

https://codepen.io/jcfranco/pen/vYmoKYq

### Reproduction Steps

1. Navigate to above link in chrome.
2. Activate the "Screen Reader" extension if insalled, if not installed, install from here https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en
3. Set focus within the preview.
4. Tab between form components and observe what the screen reader speaks

### Reproduction Version

beta.80

### Relevant Info

Window, Chrome 100.0.4896.75 (Official Build) (64-bit)

**Comment by benelan** [comment]
Note: the repro sample was made a while ago and is on `beta.62`. However it does still repro on `beta.80`.

**Comment by geospatialem** [comment]
## Current behavior
In NVDA (Windows) the `calcite-combobox-item`'s are reading back as "blank", also present on the [sample page](https://developers.arcgis.com/calcite-design-system/components/combobox/#sample).

**_Note_**: The other form elements are readable by NVDA, so it appears this applies just to the `combobox` and `combobox-items`.

## Expected behavior

### `combobox`
* Should indicate if the menu is expanded ("active" property) or collapsed.  
* The `combobox` should indicate if any items have been selected (e.g., "2 items selected")
  * Perhaps could use the `combobox-item` "selected" property?


### `combobox-item`
* When in focus the `combobox-item` should be read by name (perhaps using `value`), **and** the item number, if it is checked/unchecked, and where they are in the list (e.g., Accordion, checkbox checked, item 1 of 18)


A [hacky expected behavior via CodePen](https://codepen.io/geospatialem/pen/oNpVLYx), that includes most of the above, with the exception of the combobox-item item order (item x of x).

**Comment by geospatialem** [comment]
Was able to reproduce with JAWS. 

Some `calcite-label`'s when paired with components are not being read back to the screen reader, in particular:

- _2. Where did you hear about Calcite components?_
    -   `calcite-select`
- _3. Will you be using Calcite components in an upcoming project?_
    -   `calcite-radio-button-group`
- _4. How many components will you be using?_
    - `calcite-slider`
-   _6. How useful is the Calcite design system?_
    - `calcite-radio-group` 
- _7. What is the most important aspect(s) of a component?_
    - `calcite-checkbox`
- _9. How many stars would you give Calcite components?_
    - `calcite-rating`    

However, all can be mitigated by adding a `<fieldset>` and `<legend>`, such as:

```html
<li>
  <fieldset>
    <legend>
	    <calcite-label>How many stars would you give Calcite components?
    </legend>
	  <calcite-rating value="1"></calcite-rating>
  </calcite-label>
  </fieldset>
</li>
```

We should discuss if it we should pursue these changes for `calcite-label` or if the fieldset and legend solution is the intent moving forward.

What are your initial thoughts @jcfranco?

**Comment by geospatialem** [comment]
Repurposing this issue into two issues, this one will remain to address `combobox` and `combobox-item`, and #4600 for reading `calcite-label`.

**Comment by driskull** [comment]
I wonder if any of this has to do with having elements that are for screen readers and elements the users see and they are different.

@paulcpederson do you think this could be an issue? Maybe we need to refactor this component so that both screen readers and users are working with the same slotted elements. I don't think the w3c demo works like this.

https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-select-only.html

**Comment by paulcpederson** [comment]
@driskull I think it's not working because somebody broke it... The aria-only tree is explicitly there for screen readers. Looks like at some point the active descendant attribute was updated so that it doesn't have the prefix, so the input is pointing to an active descendent that's not there.

https://github.com/Esri/calcite-components/blob/master/src/components/combobox/combobox.tsx#L1062

Add the prefix there like `${itemUidPrefix}${this.activeDescendant}` and when the input is related correctly with the options, NVDA will read them out as you move between them (ie. "Trees, not selected, one of fifteen"). 

From what I can tell there is no way to relate two elements across the shadow dom boundary, so you have to create this screen-reader only DOM. The `calcite-combobox-item` components are in light dom, but the input is in shadow dom. Maybe NVDA/JAWS are smarter now, but last I looked at this it was impossible to related them.

**Comment by driskull** [comment]
If anyone broke it it's probably me :)

Thanks for the info!

**Comment by driskull** [comment]
@geospatialem can you test out this PR? https://github.com/Esri/calcite-components/pull/4905

**Comment by geospatialem** [comment]
The first item works great. Items are now read back via all screen readers when interacting with the combobox items. 💯

@driskull Are numbers 2 and 3 (listed below) in scope for this issue, or should separate issues be created to address them?
 
> 1. When in focus the `combobox-item` should be read by name (perhaps using `value`), **and** the item number, if it is checked/unchecked, and where they are in the list (e.g., Accordion, checkbox checked, item 1 of 18)
- [x] **Complete**


> 2. Should indicate if the [`combobox`] menu is expanded ("active" property) or collapsed.
- [ ] **Partial Completion**. There is some indication with expanded, but no indication when collapsed.


> 3. The `combobox` should indicate if any items have been selected (e.g., "2 items selected"). Perhaps could use the `combobox-item` "selected" property?
- [ ] **Not yet completed.** Unselected states are read back, but selected ones are not. If a user selects a `combobox-item`,  context is not currently provided to the screen reader that selection has taken place (unselected items are read back as unselected). Also, if the user returns to the input, currently context is not provided to the screen reader that an item has been previously selected.

**Comment by driskull** [comment]
Lets create separate issues for those @geospatialem. Thanks

**Comment by geospatialem** [comment]
Verified on `next`. Follow-up issues to follow shortly.
